### PR TITLE
:sparkles:  k/k 1.31.0 rebase

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

Rebase to 1.31.

Changes:
1. Update all external generators to match future versions of kcp
2. regression (our side kind of) fix for `genclient:skipVerbs=list,watch`. If skipped, on our end we skip List, Watch and DeleteCollections. It should be just List and Watch. Need to fix this or redo our generators
3. Make definition `def.Strict = false` as we (k/k) changed some tags where now they are `group=`, meaning its not full key-value-pair. This passes the validation as its non-strict mode. This could have consequences that we miss some API groups in the future. 

In general, we should see how to make these generators closer to upstream v2 as now they are bespoke solutions. Or live with a bespoke solution. I would better invest time in refactoring them while we have time, and it's not a hard blocker for rebase. 

## Related issue(s)

Fixes #

## Release Notes

```release-note
1.31 rebase 
Make generator stricts=false 
```

### CODE REVIEW

File: examples/pkg/apis/example/v1beta1/types.go [hunk 1]
```diff
@@ -19,7 +19,6 @@ package v1beta1
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 // +genclient
-// +genclient:skipVerbs=list,watch
```
Comment (mjudeikis) [inline examples/pkg/apis/example/v1beta1/types.go:22]: This is now somehow broken after 1.31 migrated to generics. And I'm missing historical background why this was here in the first case?

Comment (mjudeikis) [reply examples/pkg/apis/example/v1beta1/types.go:22]: I might be wrong but looks like regression in code-generator and not our code

Comment (mjudeikis) [reply examples/pkg/apis/example/v1beta1/types.go:22]: I generate clients wrong without DeleteCollections methods.

Comment (sttts) [reply examples/pkg/apis/example/v1beta1/types.go:22]: you mean they are missing?

Comment (mjudeikis) [reply examples/pkg/apis/example/v1beta1/types.go]: Fixed. I think its getting just further from upstream :)

Comment (mjudeikis) [reply examples/pkg/apis/example/v1beta1/types.go:22]: upstream client changed. Basically this skips List,Watch and DeleteCollections. And this breaks our generator a  bit.
I think we still need to refactor this to v2 generator. Not sure if it makes sense to invest now :/

**Review by embik — APPROVED** [review]

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**